x/nft/keeper/grpc_query.go

```diff
@@ -0,0 +1,190 @@
+package keeper
+
+import (
+	"context"
+
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/query"
+	"github.com/cosmos/cosmos-sdk/x/nft/types"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+var _ types.QueryServer = Keeper{}
+
+func (k Keeper) NFT(ctx context.Context,
+	request *types.QueryNFTRequest) (*types.QueryNFTResponse, error) {
+	if request == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	nft, has := k.GetNFT(sdkCtx, request.Type, request.Id)
+	if !has {
+		return nil, status.Errorf(codes.InvalidArgument, "not found nft")
+	}
+	return &types.QueryNFTResponse{NFT: &nft}, nil
+}
+
+func (k Keeper) NFTs(ctx context.Context,
+	request *types.QueryNFTsRequest) (*types.QueryNFTsResponse, error) {
+	if request == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+
+	var nfts []*types.NFT
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	if len(request.Owner) == 0 {
+		store := sdkCtx.KVStore(k.storeKey)
+		typeStore := prefix.NewStore(store, types.NFTKey)
+		pageRes, err := query.Paginate(typeStore, request.Pagination, func(_, value []byte) error {
+			var nft types.NFT
+			err := k.cdc.Unmarshal(value, &nft)
+			if err != nil {
+				return err
+			}
+			nfts = append(nfts, &nft)
+			return nil
+		})
+		if err != nil {
+			return nil, status.Errorf(codes.InvalidArgument, "paginate: %v", err)
+		}
+		return &types.QueryNFTsResponse{
+			NFTs:       nfts,
+			Pagination: pageRes,
+		}, nil
+	}
+
+	owner, err := sdk.AccAddressFromBech32(request.Owner)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "invalid address: %s", err.Error())
+	}
+
+	balances := k.bk.GetAllBalances(sdkCtx, owner)
```

Comment:
we should also use a pagination when `owner` is specified. There can be an owner with thousands of NFTs.